src/sds_data_model/_raster.py

```diff
@@ -0,0 +1,466 @@
+    """
+    return DataArray(
+        data=data,
+        coords={
+            data_array.rio.grid_mapping: data_array[
+                data_array.rio.grid_mapping
+            ].rio.update_attrs({"GeoTransform": geotransform}),
+        },
+        dims=("northings", "eastings"),
+        attrs={"nodata": data_array.rio.nodata},
+    ).rename(data_array.name)
+
+
+def _get_resample_shape(
+    data_array: DataArray,
+    expected_cell_size: int,
+) -> NDArray:
+    """Determine the new (scaled) shape of an input DataArray from a cell_size value.
+
+    Args:
+        data_array (DataArray): Input `DataArray`.
+        expected_cell_size (int): New cell size for data. Acts as the resampling factor.
+
+    Returns:
+        NDArray: 2d array of (height, width).
+    """
+    return (
+        (
+            (
+                array([data_array.rio.transform().a, -data_array.rio.transform().e])
+                / expected_cell_size
+            )
+            * data_array.shape
+        )
+        .round()
+        .astype("int")
+    )
+
+
+def _resample_cell_size(
+    data_array: DataArray,
+    expected_cell_size: int,
+    categorical: bool,
+) -> DataArray:
+    """Returns a `DataArray` with scaled/resampled data using OpenCV.
+
+    Args:
+        data_array (DataArray): Input `DataArray`.
+        expected_cell_size (int): New cell size for data. Acts as the resampling factor.
+        categorical (bool): Determines which resampling/interpolation is used.
+            Set to False will use bilinear. Set to True will use nearest-neighbour.
+
+    Returns:
+        DataArray
+    """
+    if categorical:
+        interpolation = INTER_NEAREST
+    else:
+        interpolation = INTER_LINEAR
+    resampled = resize(
+        src=data_array.data,
+        dsize=_get_resample_shape(data_array, expected_cell_size)[::-1],
+        interpolation=interpolation,
+    )
+
+    return _create_data_array(
+        data_array=data_array,
+        data=resampled,
+        geotransform=(
+            f"{data_array.rio.transform().c} {expected_cell_size} 0 "
+            f"{data_array.rio.transform().f} 0 -{expected_cell_size}"
+        ),
+    )
+
+
+def _check_no_data(
+    data_array: DataArray,
+    nodata: Optional[float],
+) -> None:
+    """Checks and conditionally writes a `nodata` value to data_array.
+
+    Args:
+        data_array (DataArray): Input `DataArray`.
+        nodata (Optional[float]): Value that will be written to the
+            input `DataArray` if it is not `None`.
+
+    Raises:
+        Exception: If the `nodata` arg is `None` and the input `DataArray`
+            does not have a nodata value (that is not equal to None).
+    """
+    if nodata is not None:
+        data_array.rio.write_nodata(nodata, inplace=True)
+    elif data_array.rio.nodata is None:
+        raise Exception(
+            "Input dataset does not have a nodata value. One must be provided."
+        )
```

Comment:
You should always try and use the most precise exception possible which, in this case, I think is [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError). 

```suggestion
    Raises:
        ValueError: If the `nodata` arg is `None` and the input `DataArray`
            does not have a nodata value (that is not equal to None).
    """
    if nodata is not None:
        data_array.rio.write_nodata(nodata, inplace=True)
    elif data_array.rio.nodata is None:
        raise ValueError(
            "Input dataset does not have a nodata value. One must be provided."
        )
```